crates/red_knot_python_semantic/src/types.rs

```diff
@@ -574,8 +574,11 @@ impl<'db> Type<'db> {
         Self::BytesLiteral(BytesLiteralType::new(db, bytes))
     }
 
-    pub fn tuple(db: &'db dyn Db, elements: &[Type<'db>]) -> Self {
-        Self::Tuple(TupleType::new(db, elements))
+    pub fn tuple<T: Into<Type<'db>>>(
+        db: &'db dyn Db,
+        elements: impl IntoIterator<Item = T>,
+    ) -> Self {
+        TupleType::from_elements(db, elements)
     }
```

Comment:
nit: this method now feels a bit redundant. I'd probably have got rid of it and changed the callsites to use the new `TupleType::from_elements()` method